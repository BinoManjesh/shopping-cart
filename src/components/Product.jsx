import { shape, string, number, node } from "prop-types";
import styles from "../styles/Product.module.css";

Product.propTypes = {
  product: shape({
    title: string,
    price: number,
    image: string,
    rating: shape({
      rate: number,
      count: number,
    }),
  }),
  children: node,
};

function Product({ product, children }) {
  return (
    <div className={styles.product}>
      <img className={styles.img} src={product.image} alt={product.title} />
      <h2>{product.title}</h2>
      <p>
        {product.rating.rate}
        <span className={styles.star}>â˜…</span> ({product.rating.count})
      </p>
      <p>$ {product.price}</p>
      <div>{children}</div>
    </div>
  );
}

export default Product;
