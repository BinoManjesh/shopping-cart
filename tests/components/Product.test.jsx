import { render } from "@testing-library/react";
import Product from "../../src/components/Product";

describe("Product component", () => {
  it("matches snapshot", () => {
    const product = {
      title: "title",
      price: 20.5,
      image: ".",
      rating: {
        rate: 4.9,
        count: 23423,
      },
    };

    const { container } = render(<Product product={product} />);
    expect(container).toMatchSnapshot();
  });
});
