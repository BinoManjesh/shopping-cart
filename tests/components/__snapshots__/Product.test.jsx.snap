// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product component > matches snapshot 1`] = `
<div>
  <div
    class="_product_b0e857"
  >
    <img
      alt="title"
      class="_img_b0e857"
      src="."
    />
    <h2>
      title
    </h2>
    <p>
      4.9
      <span
        class="_star_b0e857"
      >
        â˜…
      </span>
       (
      23423
      )
    </p>
    <p>
      $ 
      20.5
    </p>
    <div />
  </div>
</div>
`;
